fnE = function()
  error('oooops')
end
